front/app/containers/Admin/reporting/components/ReportBuilderPage/EmptyState.tsx

```diff
@@ -27,6 +34,14 @@ interface Props {
 }
 
 const EmptyState = ({ onOpenModal }: Props) => {
+  const appConfig = useAppConfiguration();
+
+  if (isNilOrError(appConfig)) {
+    return null;
+  }
+  const isReportBuilderAllowed =
+    appConfig.attributes.settings.report_builder?.allowed;
```

Comment:
You can use useFeatureFlag for this:

```suggestion
  const isReportBuilderAllowed = useFeatureFlag({
    name: 'report_builder',
    onlyCheckAllowed: true,
  });
```